tools/generate-helpers/pg-table-bindings/store_test.go.tpl

```diff
@@ -98,9 +128,20 @@ func (s *{{$namePrefix}}StoreSuite) TestStore() {
 	s.NoError(err)
 	s.True({{$name}}Exists)
 	s.NoError(store.Upsert(ctx, {{$name}}))
-    {{- if or (.Obj.IsGloballyScoped) (.Obj.HasPermissionChecker) }}
+    {{- if or (.Obj.IsGloballyScoped) (.Obj.HasPermissionChecker) (.Obj.IsDirectlyScoped)}}
 	s.ErrorIs(store.Upsert(withNoAccessCtx, {{$name}}), sac.ErrResourceAccessDenied)
     {{- end }}
+	{{- if (.Obj.IsDirectlyScoped)}}
+	s.ErrorIs(store.Upsert(withNoAccessToClusterCtx, {{ $name }}), sac.ErrResourceAccessDenied)
+	s.ErrorIs(store.Upsert(withAccessToDifferentNsCtx, {{ $name }}), sac.ErrResourceAccessDenied)
+	s.NoError(store.Upsert(withAccessCtx, {{ $name }}))
+	s.NoError(store.Upsert(withAccessToClusterCtx, {{ $name }}))
+	s.NoError(store.Upsert(withAccessToClusterCtx, {{ $name }}))
+	s.ErrorIs(store.UpsertMany(withAccessToDifferentNsCtx, []*{{.Type}}{ {{ $name }} }), sac.ErrResourceAccessDenied)
+	s.ErrorIs(store.UpsertMany(withNoAccessToClusterCtx, []*{{.Type}}{ {{ $name }} }), sac.ErrResourceAccessDenied)
+	s.NoError(store.UpsertMany(withAccessCtx, []*{{.Type}}{ {{ $name }} }))
+	s.NoError(store.UpsertMany(withAccessToClusterCtx, []*{{.Type}}{ {{ $name }} }))
+	{{- end }}
```

Comment:
I like the comprehensive test here.
We may want to have a dedicated scoped access control test file next to store_test where the behaviour of each function is tested in more details with regards to the various possible scopes present in the context.

Reply:
Sure I'll do it in a separated PR for [ROX-10072](https://issues.redhat.com/browse/ROX-10072)